src/TestFramework/Coverage/CoverageChecker.php

```diff
@@ -222,4 +217,11 @@ private function isPcovIncludedInInitialTestPhpOptions(): bool
             $this->initialTestPhpOptions
         );
     }
+
+    private function getCoverageIndexFilePath(): string
+    {
+        return Path::canonicalize(
+            (new IndexXmlCoverageReader($this->coveragePath))->getIndexXmlPath()
```

Comment:
I feel this `IndexXmlCoverageReader` should be injected instead. Also since you are moving this in a method, might be worth memoizing that call

Reply:
Yes, it'll good to inject it.